src/keygen/participant.rs

```diff
@@ -13,43 +13,55 @@ use crate::{
         keygen_commit::{KeygenCommit, KeygenDecommit},
         keyshare::{KeySharePrivate, KeySharePublic},
     },
+    local_storage::{LocalStorage, TypeTag},
     messages::{KeygenMessageType, Message, MessageType},
     participant::{Broadcast, ProcessOutcome, ProtocolParticipant},
     protocol::ParticipantIdentifier,
     run_only_once,
-    storage::{PersistentStorageType, Storable, Storage},
+    storage::{PersistentStorageType, Storage},
     utils::k256_order,
     zkp::pisch::{PiSchInput, PiSchPrecommit, PiSchProof, PiSchSecret},
     CurvePoint,
 };
 use libpaillier::unknown_order::BigNumber;
 use merlin::Transcript;
 use rand::{CryptoRng, RngCore};
-use serde::{Deserialize, Serialize};
 use tracing::{info, instrument};
 
-// Storage identifiers for the keygen protocol.
-#[derive(Clone, Copy, Debug, Serialize)]
-#[serde(tag = "KeyGen")]
-enum StorageType {
-    Ready,
-    Commit,
-    Decommit,
-    SchnorrPrecom,
-    GlobalRid,
+struct StorageTypeReady;
```

Comment:
question: Do you think these structs will have the same serde-overlapping-names problem that we had with the enum? E.g. if we also have a `StorageTypeReady` in auxinfo.

Reply:
No, because it's entirely type based (there's no serialization being done). That's one of the many benefits of this approach :-)